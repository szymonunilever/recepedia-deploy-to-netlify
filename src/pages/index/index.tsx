import { graphql } from 'gatsby';
import React from 'react';
import Layout from 'src/components/Layout/Layout';
import SEO from 'src/components/Seo/Seo';
import { Text, TagName } from 'src/components/lib/components/Text';

import { RecipeListing } from 'src/components/lib/components/RecipeListing';
import Hero from 'src/components/lib/components/Hero';
import { findPageComponentContent } from 'src/utils';
import { RecipeItem } from 'src/components/lib/components/RecipeListing/partials';

const HomePage = ({ data }: HomePageProps) => {
  const page = data.allPage.edges[0].node;
  const components = page.components.items;
  const recipes = data.allRecipe.nodes;

  return (
    <Layout>
      <SEO title="Recepedia Home" />

      <Text tag={TagName['h1']} text={data.allPage.edges[0].node.title} />
      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'LatestAndGreatest'
          )}
          list={recipes}
        />
      </section>

      <section>
        <RecipeListing
          content={findPageComponentContent(
            components,
            'RecipeListing',
            'TopRecipes'
          )}
          list={recipes}
        />
      </section>

      <section>
        <Hero
          content={findPageComponentContent(components, 'Hero')}
          viewType="Image"
        />
      </section>
    </Layout>
  );
};

export default HomePage;

export const pageQuery = graphql`
  {
    allPage(filter: { type: { eq: "Home" } }) {
      edges {
        node {
          ...PageFields
        }
      }
    }
    allRecipe(skip: 10) {
      nodes {
        ...RecipeFields
      }
    }
  }
`;

interface HomePageProps {
  data: {
    allPage: {
      edges: Edge<PageNode>[];
    };
    allRecipe: {
      nodes: RecipeItem[];
    };
  };
}

interface Edge<T> {
  node: T;
}

interface PageNode {
  components: {
    items: {
      name: string;
      content: any;
    }[];
  };
  title: string;
  type: string;
}
