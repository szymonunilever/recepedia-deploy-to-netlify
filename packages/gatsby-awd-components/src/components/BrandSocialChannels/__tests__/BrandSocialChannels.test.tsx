import { mount, ReactWrapper } from 'enzyme';
import React from 'react';
import { BrandSocialChannels } from '../index';
import toJson from 'enzyme-to-json';

describe('<BrandSocialChannels />', () => {
  let wrapper: ReactWrapper;
  const mock = {
    socialItems: {
      facebook: {
        label: 'Facebook',
        url: 'https://www.facebook.com/',
      },
      instagram: {
        label: 'Instagram',
        url: 'https://www.instagram.com/',
      },
      twitter: {
        label: 'Twitter',
        url: 'https://twitter.com/',
      },
      pinterest: {
        label: 'Pinterest',
        url: 'https://www.pinterest.com/',
      },
    },
  };
  const props: any = {
    listIcons: {
      facebook: <div>Facebook</div>,
      instagram: <div>Instagram</div>,
    },
    content: mock,
  };

  beforeEach(() => {
    wrapper = mount(<BrandSocialChannels {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('matches the snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
