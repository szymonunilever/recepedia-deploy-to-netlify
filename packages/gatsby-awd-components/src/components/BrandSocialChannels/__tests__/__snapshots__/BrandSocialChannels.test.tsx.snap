// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrandSocialChannels /> matches the snapshot 1`] = `
<BrandSocialChannels
  content={
    Object {
      "socialItems": Object {
        "facebook": Object {
          "label": "Facebook",
          "url": "https://www.facebook.com/",
        },
        "instagram": Object {
          "label": "Instagram",
          "url": "https://www.instagram.com/",
        },
        "pinterest": Object {
          "label": "Pinterest",
          "url": "https://www.pinterest.com/",
        },
        "twitter": Object {
          "label": "Twitter",
          "url": "https://twitter.com/",
        },
      },
    }
  }
  listIcons={
    Object {
      "facebook": <div>
        Facebook
      </div>,
      "instagram": <div>
        Instagram
      </div>,
    }
  }
>
  <ul
    className="socialChannels social-channels"
    data-componentname="social-channels"
    data-componentview="social-channels"
  >
    <SocialChannel
      attributes={
        Object {
          "aria-label": "facebook",
        }
      }
      icon={
        <div>
          Facebook
        </div>
      }
      key="facebook"
      label={false}
      url="https://www.facebook.com/"
    >
      <li
        className="socialChannels__item social-channels__item"
      >
        <a
          aria-label="facebook"
          className="socialChannels__link social-channels__link"
          href="https://www.facebook.com/"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          <div
            className="socialChannels__icon social-channels__icon"
          >
            <div>
              Facebook
            </div>
          </div>
        </a>
      </li>
    </SocialChannel>
    <SocialChannel
      attributes={
        Object {
          "aria-label": "instagram",
        }
      }
      icon={
        <div>
          Instagram
        </div>
      }
      key="instagram"
      label={false}
      url="https://www.instagram.com/"
    >
      <li
        className="socialChannels__item social-channels__item"
      >
        <a
          aria-label="instagram"
          className="socialChannels__link social-channels__link"
          href="https://www.instagram.com/"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          <div
            className="socialChannels__icon social-channels__icon"
          >
            <div>
              Instagram
            </div>
          </div>
        </a>
      </li>
    </SocialChannel>
  </ul>
</BrandSocialChannels>
`;
